Makefile

```diff
@@ -0,0 +1,120 @@
+.DEFAULT_GOAL	:= compile
+
+#------------------------------------------------------------------------------
+# Variables
+#------------------------------------------------------------------------------
+
+SHELL 		:= /bin/bash
+BINDIR		:= bin
+BUILDDIR    := build
+DOCKERDIR	:= docker
+RELEASEDIR  := release
+OUTPUT_NAME := envoyctl
+GOHOSTOS 	:= $(shell go env GOHOSTOS)
+
+ifndef GOOS
+    GOOS := $(GOHOSTOS)
+endif
+
+ifndef GOARCH
+	GOARCH := $(shell go env GOHOSTARCH)
+endif
+
+GOFILES		= $(shell find . -type f -name '*.go' -not -path "./vendor/*")
+GODIRS		= $(shell go list -f '{{.Dir}}' ./... | grep -vFf <(go list -f '{{.Dir}}' ./vendor/...))
+GOPKGS		= $(shell go list ./... | grep -vFf <(go list ./vendor/...))
+
+APP_VER		:= $(shell git describe --always 2> /dev/null || echo "unknown")
+
+# linker flags to set build info variables
+BUILD_SYM	:= github.com/envoyproxy/go-control-plane/pkg/version
+LDFLAGS		+= -X $(BUILD_SYM).version=$(APP_VER)
+LDFLAGS		+= -X $(BUILD_SYM).gitRevision=$(shell git rev-parse --short HEAD 2> /dev/null  || echo unknown)
+LDFLAGS		+= -X $(BUILD_SYM).branch=$(shell git rev-parse --abbrev-ref HEAD 2> /dev/null  || echo unknown)
+LDFLAGS		+= -X $(BUILD_SYM).buildUser=$(shell whoami || echo nobody)@$(shell hostname -f || echo builder)
+LDFLAGS		+= -X $(BUILD_SYM).buildDate=$(shell date +%Y-%m-%dT%H:%M:%S%:z)
+LDFLAGS		+= -X $(BUILD_SYM).goVersion=$(word 3,$(shell go version))
+
+.PHONY: compile
+compile:
+	@go build -ldflags '$(LDFLAGS)' -o $(BINDIR)/$(OUTPUT_NAME)
+
+.PHONY: proto
+proto:
+	@go get -u github.com/golang/protobuf/{proto,protoc-gen-go}
+	@protoc --go_out=plugins=grpc:pkg/ vendor/github.com/envoyproxy/data-plane-api/api/*.proto \
+					--proto_path=vendor/github.com/envoyproxy/data-plane-api \
+					--proto_path=vendor/github.com/googleapis/googleapis/
+	@protoc --go_out=plugins=grpc:pkg/ vendor/github.com/envoyproxy/data-plane-api/api/filter/*.proto \
```

Comment:
I don't think this is right. When I run this command I get a file with imports like this
```
package envoy_api_v2_filter

import proto "github.com/golang/protobuf/proto"
import fmt "fmt"
import math "math"
import envoy_api_v21 "api"
import envoy_api_v22 "api"
import envoy_api_v24 "api"
```
These should be `github.com/envoyproxy/go-control-plane/pkg/envoy/api`

Reply:
Looks like the same issue as https://github.com/golang/protobuf/issues/63

Reply:
Thanks for catching this @davecheney .. I think I will take this line out for now, since @kyessenov is generating protos using bazel (using same proto version as envoy repo), and our plan is to commit these protos into the repo, so that others can just import this library.

Reply:
ref https://github.com/envoyproxy/data-plane-api/pull/204